### PR TITLE
[Feature][Plugin] Add seatunnel-connector-spark-sink-mongodb

### LINKED ISSUE #1084: [Feature][Plugin] Add seatunnel-connector-spark-sink-mongodb
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Add seatunnel-connector-spark-sink-mongodb

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
## Purpose of this pull request

close #1084
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by CalvinKirs** [comment]
hi, please confirm check list

File: docs/en/spark/configuration/sink-plugins/MongoDB.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Sink plugin common parameters, please refer to [Sink Plugin](./sink-plugin.md) for details
+
+## Examples
+
+```bash
+mongodb {
+    writeconfig.uri = "mongodb://username:password@127.0.0.1:27017/test_db"
+    writeconfig.database = "test_db"
+    writeconfig.collection = "test_collection"
+    result_table_name = "mongodb_result_table"
```
Comment (RickyHuo) [inline docs/en/spark/configuration/sink-plugins/MongoDB.md]: sink doesn't need `result_table_name`

File: seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        val value = String.valueOf(entry.getValue.unwrapped())
+        map.put(key, value)
+      })
+    writeConfig = WriteConfig(map)
+  }
+
+  override def checkConfig(): CheckResult = {
+    if (TypesafeConfigUtils.hasSubConfig(config, confPrefix)) {
+      val read = TypesafeConfigUtils.extractSubConfig(config, confPrefix, false)
+      if (read.hasPath("uri") && read.hasPath("database") && read.hasPath("collection")) {
```
Comment (RickyHuo) [inline seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala]: We can use common utils of `CheckConfigUtil` and static func of `CheckResult.error` now :)

**Review by RickyHuo — CHANGES_REQUESTED** [review]

File: docs/en/spark/configuration/sink-plugins/MongoDB.md (hunk 1 above)
Comment (wuchunfu) [reply docs/en/spark/configuration/sink-plugins/MongoDB.md]: ok, I will fix it.

File: seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala (hunk 2 above)
Comment (wuchunfu) [reply seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala]: ok, I will fix it.

**Comment by wuchunfu** [comment]
cc @CalvinKirs @RickyHuo

File: docs/en/spark/configuration/sink-plugins/MongoDB.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Write data to `MongoDB`
+
+## Options
+
+| name                   | type   | required | default value |
+| ---------------------- | ------ | -------- | ------------- |
+| writeconfig.uri        | string | yes      | -             |
+| writeconfig.database   | string | yes      | -             |
+| writeconfig.collection | string | yes      | -             |
```
Comment (RickyHuo) [inline docs/en/spark/configuration/sink-plugins/MongoDB.md:13]: `writeconfig.*`  should be listed.

File: seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala (hunk 2 above)
Comment (RickyHuo) [inline seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala]: How about this below ?

> CheckConfigUtil.check("writeconfig.uri", "writeconfig.database", "writeconfig.collection")

**Review by RickyHuo — CHANGES_REQUESTED** [review]

File: docs/en/spark/configuration/sink-plugins/MongoDB.md (hunk 3 above)
Comment (wuchunfu) [reply docs/en/spark/configuration/sink-plugins/MongoDB.md:13]: ok, thanks

File: seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala (hunk 2 above)
Comment (wuchunfu) [reply seatunnel-connectors/seatunnel-connector-spark-mongodb/src/main/scala/org/apache/seatunnel/spark/sink/MongoDB.scala]: This looks more elegant

**Review by RickyHuo — APPROVED** [review]
LGTM

**Pull request merged**